scripts/sync-locale.ts

```diff
@@ -0,0 +1,243 @@
+    }
+  }
+
+  return result
+}
+
+function addTranslationComments(obj: TranslatePair): TranslatePair {
+  return Object.entries(obj).reduce((all, [key, value]) => {
+    if (isArray(value)) {
+      all[key] = value.map((item) => `${item} ${commentInfo}`)
+    } else if (isObject(value)) {
+      all[key] = addTranslationComments(value)
+    } else if (isString(value)) {
+      all[key] = `${value} ${commentInfo}`
+    }
+
+    return all
+  }, {} as TranslatePair)
+}
+
+function objectToTypescript(obj: TranslatePair, indent = 0): string {
+  const spaces = '  '.repeat(indent)
+  const items: string[] = []
+
+  Object.entries(obj).forEach(([key, value]) => {
+    if (isArray(value)) {
+      let needComment = false
+      const arrayItems = value.map((item) => {
+        if (item.includes(commentInfo)) {
+          const [actualValue] = item.split(commentInfo)
+          const escapedValue = actualValue.trim().replace(/'/g, "\\'")
+
+          needComment = true
+          return `'${escapedValue}'`
+        } else {
+          const escapedValue = item.replace(/'/g, "\\'")
+          return `'${escapedValue}'`
+        }
+      })
+
+      items.push(
+        `${spaces}  ${key}: [${arrayItems.join(', ')}],${
+          needComment ? ` ${commentInfo}` : ''
+        }`
+      )
+    } else if (isObject(value)) {
+      items.push(
+        `${spaces}  ${key}: {\n${objectToTypescript(
+          value as TranslatePair,
+          indent + 1
+        )}\n${spaces}  },`
+      )
+    } else if (isString(value)) {
+      if (value.includes(commentInfo)) {
+        const [actualValue] = value.split(commentInfo)
+        const escapedValue = actualValue.trim().replace(/'/g, "\\'")
+        items.push(`${spaces}  ${key}: '${escapedValue}', ${commentInfo}`)
+      } else {
+        const escapedValue = value.replace(/'/g, "\\'")
+        items.push(`${spaces}  ${key}: '${escapedValue}',`)
+      }
+    }
+  })
+
+  return items.join('\n')
+}
+
+function generateTypescriptFile(obj: TranslatePair): string {
+  return `export default {\n${objectToTypescript(obj)}\n}\n`
+}
+
+function countKeys(obj: TranslatePair): number {
+  return Object.values(obj).reduce((all, value) => {
+    if (isObject(value) && !isArray(value)) {
+      all += countKeys(value)
+    } else {
+      all++
+    }
+
+    return all
+  }, 0)
+}
+
+function main() {
+  consola.start('Starting to synchronize content from English files...')
+
+  const enContent = readFileSync(enFile, 'utf-8')
+  const enObject = parseLocaleFile(enContent)
+  const localeFiles = glob.sync(localePath)
+  let totalUpdated = 0
+  let totalAdded = 0
+
+  consola.info(`Found ${localeFiles.length} locale files to process`)
+
+  localeFiles.forEach((filePath) => {
+    const fileName = basename(filePath)
+
+    consola.start(`Processing ${fileName}...`)
+
+    try {
+      const content = readFileSync(filePath, 'utf-8')
+      const targetObject = parseLocaleFile(content)
```

Comment:
Maybe we can load the file content directly through `import(filePath)`.


```suggestion
      const targetObject = (await import(filePath)).default
```

Reply:
Indeed, I seem to have missed the logic. You deleted all the fields to be translated in the `parseLocaleFile` function.